pkg/controller/install/install_controller.go

```diff
@@ -113,54 +119,88 @@ func (r *ReconcileInstall) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+// Initialize status conditions
+func (r *ReconcileInstall) initStatus(instance *eventingv1alpha1.Install) error {
+	if len(instance.Status.Conditions) == 0 {
+		instance.Status.InitializeConditions()
+		if err := r.updateStatus(instance); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Update the status subresource
+func (r *ReconcileInstall) updateStatus(instance *eventingv1alpha1.Install) error {
+
+	// Account for https://github.com/kubernetes-sigs/controller-runtime/issues/406
+	gvk := instance.GroupVersionKind()
+	defer instance.SetGroupVersionKind(gvk)
+
+	if err := r.client.Status().Update(context.TODO(), instance); err != nil {
+		return err
+	}
+	return nil
+}
+
 // Apply the embedded resources
 func (r *ReconcileInstall) install(instance *eventingv1alpha1.Install) error {
 	// Transform resources as appropriate
-	fns := []mf.Transformer{mf.InjectOwner(instance), addSCCforSpecialClusterRoles}
-	if len(instance.Spec.Namespace) > 0 {
-		fns = append(fns, mf.InjectNamespace(instance.Spec.Namespace))
+	fns := []mf.Transformer{
+		mf.InjectOwner(instance),
+		mf.InjectNamespace(instance.GetNamespace()),
+		addSCCforSpecialClusterRoles,
 	}
 	r.config.Transform(fns...)
 
-	if instance.Status.Version == version.Version {
-		// we've already successfully applied our YAML
+	if instance.Status.IsDeploying() {
 		return nil
 	}
+	defer r.updateStatus(instance)
+
 	// Apply the resources in the YAML file
 	if err := r.config.ApplyAll(); err != nil {
+		instance.Status.MarkInstallFailed(err.Error())
 		return err
 	}
 
 	// Update status
-	instance.Status.Resources = r.config.Resources
 	instance.Status.Version = version.Version
-	if err := r.client.Status().Update(context.TODO(), instance); err != nil {
-		return err
-	}
+	instance.Status.MarkInstallSucceeded()
 	return nil
 }
 
-func autoInstall(c client.Client, ns string) (err error) {
-	const path = "deploy/crds/eventing_v1alpha1_install_cr.yaml"
-	log.Info("Automatic Install requested", "namespace", ns)
-	installList := &eventingv1alpha1.InstallList{}
-	err = c.List(context.TODO(), &client.ListOptions{Namespace: ns}, installList)
-	if err != nil {
-		log.Error(err, "Unable to list Installs")
-		return err
+// Check for all deployments available
+// TODO: what about statefulsets?
```

Comment:
Just a note on this:

the statefulsets are only things from `knative-source` NS (`eventing-sources` repo).

For 0.6.0 these are no concern here anymore . (e.g. some of the things that got move out of "sources" to "core" are now deployments